### ISSUE TITLE
brew install ngspice fails with undefined symbol _rl_catch_signals

### ISSUE THREAD

**Issue by nilnullzip** [Homebrew/homebrew-core#32972]
- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?

https://gist.github.com/629b805f07879bb86a95c8aa91ab478b

- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
- What you expected to happen

I was trying to install ngspice. `brew install` fails with undefined symbol `_rl_catch_signals`

- Step-by-step reproduction instructions (by running `brew install` commands)

```
brew install ngspice
```

- What happened (include command output)

```
$ brew install ngspice
==> Downloading https://downloads.sourceforge.net/project/ngspice/ng-spice-rework/28/ngspice-28.tar.g
Already downloaded: /Users/juan/Library/Caches/Homebrew/downloads/f6a60e8d950d16ce3a39ac0c3ba87fc3e70834d2ffb54cad49dfa561532c9918--ngspice-28.tar.gz
==> ./configure --prefix=/usr/local/Cellar/ngspice/28 --with-editline=yes --enable-xspice --without-x
==> make install
Last 15 lines from /Users/juan/Library/Logs/Homebrew/ngspice/02.make:
  CC       ngspice-ngspice.o
rm -f spinit spinit.tmp
srcdir=''; \
	  test -f ./spinit.in || srcdir=./; \
	  sed -e 's|@XSPICEINIT[@]||g' -e 's|@pkglibdir[@]|/usr/local/Cellar/ngspice/28/lib/ngspice|g' ${srcdir}spinit.in >spinit.tmp
mv spinit.tmp spinit
  CCLD     ngspice
Undefined symbols for architecture x86_64:
  "_rl_catch_signals", referenced from:
      _app_rl_readlines in ngspice-main.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [ngspice] Error 1
make[1]: *** [install-recursive] Error 1
make: *** [install-recursive] Error 1

READ THIS: https://docs.brew.sh/Troubleshooting
```

**Comment by nilnullzip** [comment]
I believe this is the result of formula using `--with-editline=yes` It seems `editline` is missing `rl_catch_signals`. I suspect the formula should be using `--with-readline=yes`. I can't see how this formula ever worked.

**Comment by fxcoudert** [comment]
Fixed in #33066
